lib/sync-records.ts

```diff
@@ -67,98 +67,136 @@ export const syncWorkbookRecords = async ({
-          scheduledWeeklyHours: r.values.scheduledWeeklyHours.value as number,
-          flatfileRecordId: r.id,
-          jobId: job.id,
-        };
-
-        if (
-          r.values.managerId.value &&
-          (r.values.managerId.value as string).length > 0
-        ) {
-          try {
-            // Does the manager record already exist?
-            let manager = await prismaClient.employee.findUnique({
-              where: {
-                organizationId_employeeId: {
-                  organizationId,
-                  employeeId: r.values.managerId.value as string,
-                },
-              },
-            });
-
-            if (manager) {
-              data = { ...data, managerId: manager.id };
-            }
-          } catch (error) {
-            console.error(
-              "Error - managerId not found:",
-              r.values.managerId.value
+          if (!job) {
+            throw new Error(
+              `Error: job not found for employeeId ${
+                r.values.employee.value
+              }, jobName: ${values.jobName.value as string}`
             );
           }
+
+          const hireDate = parseDate({
+            fieldName: "hireDate",
+            value: values.hireDate.value as string,
+          });
+
+          let data: Parameters<typeof upsertEmployee>[0] = {
+            organizationId,
+            employeeId: r.values.employeeId.value as string,
+            firstName: r.values.firstName.value as string,
+            lastName: r.values.lastName.value as string,
+            hireDate: hireDate.toJSDate(),
+            endEmploymentDate: r.values.endEmploymentDate.value
+              ? parseDate({
+                  fieldName: "endEmploymentDate",
+                  value: values.endEmploymentDate.value as string,
+                }).toJSDate()
+              : null,
+            positionTitle: r.values.positionTitle.value as string,
+            employeeTypeId,
+            defaultWeeklyHours: r.values.defaultWeeklyHours.value as number,
+            scheduledWeeklyHours: r.values.scheduledWeeklyHours.value as number,
+            flatfileRecordId: r.id,
+            jobId: job.id,
+          };
+
+          const recordManagerId = r.values.managerId.value as string;
+
+          await upsertEmployeeAddingManager(data, recordManagerId);
+        } catch (error) {
+          console.error(
+            `Error: syncing employee record for user ${userId}, record ${r.id}`,
+            error
+          );
         }
+      }
+    };
 
-        await upsertEmployee(data);
-      } catch (error) {
-        console.error(
-          `Error: syncing employee record for user ${userId}, record ${r.id}`,
-          error
-        );
+    await upsertEmployeesWithManager();
+
+    let retriesAttempted = 0;
+    const retryUpsert = async (retriesScheduled: number) => {
+      while (retriesAttempted < retriesScheduled) {
```

Comment:
Make the gating variable very clear to avoid infinite loops.

e.g. 

const MAX_RETRY_ATTEMPTS=4;

while ( retriesAttempted < MAX_RETRY_ATTEMPTS)

Reply:
Edited and test ok